alacritty/src/renderer/mod.rs

```diff
@@ -1123,37 +1138,42 @@ fn load_glyph(
 }
 
 #[inline]
-fn clear_atlas(atlas: &mut Vec<Atlas>, current_atlas: &mut usize) {
-    for atlas in atlas.iter_mut() {
-        atlas.clear();
+fn clear_atlas(layers: &mut Vec<Layer>) {
+    for layer in layers.iter_mut() {
+        for atlas in layer.atlas.iter_mut() {
+            atlas.clear();
+        }
+        layer.current_atlas = 0;
     }
-    *current_atlas = 0;
 }
 
 impl<'a> LoadGlyph for LoaderApi<'a> {
     fn load_glyph(&mut self, rasterized: &RasterizedGlyph) -> Glyph {
-        load_glyph(self.active_tex, self.atlas, self.current_atlas, rasterized)
+        load_glyph(self.layers, rasterized)
     }
 
     fn clear(&mut self) {
-        clear_atlas(self.atlas, self.current_atlas)
+        clear_atlas(self.layers)
     }
 }
 
 impl<'a, C> LoadGlyph for RenderApi<'a, C> {
     fn load_glyph(&mut self, rasterized: &RasterizedGlyph) -> Glyph {
-        load_glyph(self.active_tex, self.atlas, self.current_atlas, rasterized)
+        load_glyph(self.layers, rasterized)
     }
 
     fn clear(&mut self) {
-        clear_atlas(self.atlas, self.current_atlas)
+        clear_atlas(self.layers)
     }
 }
 
 impl<'a, C> Drop for RenderApi<'a, C> {
     fn drop(&mut self) {
-        if !self.batch.is_empty() {
-            self.render_batch();
+        if !self.layers[0].batch.is_empty() {
+            self.render_batch(LayerType::RegularGlyphs);
+        }
+        if !self.layers[1].batch.is_empty() {
+            self.render_batch(LayerType::ColoredGlyphs);
         }
```

Comment:
This doesn't seem very extensible to me. We should iterate over the layers here in a generic fashion rather than having to hard-code all layers manually.